beacon_node/http_api/src/lib.rs

```diff
@@ -2385,6 +2400,177 @@ pub fn serve<T: BeaconChainTypes>(
      * beacon/rewards
      */
 
+    let beacon_light_client_path = eth_v1
+        .and(warp::path("beacon"))
+        .and(warp::path("light_client"))
+        .and(chain_filter.clone());
+
+    // GET beacon/light_client/bootrap/{block_root}
+    let get_beacon_light_client_bootstrap = beacon_light_client_path
+        .clone()
+        .and(task_spawner_filter.clone())
+        .and(warp::path("bootstrap"))
+        .and(warp::path::param::<Hash256>().or_else(|_| async {
+            Err(warp_utils::reject::custom_bad_request(
+                "Invalid block root value".to_string(),
+            ))
+        }))
+        .and(warp::path::end())
+        .and(warp::header::optional::<api_types::Accept>("accept"))
+        .then(
+            |chain: Arc<BeaconChain<T>>,
+             task_spawner: TaskSpawner<T::EthSpec>,
+             block_root: Hash256,
+             accept_header: Option<api_types::Accept>| {
+                task_spawner.blocking_response_task(Priority::P1, move || {
+                    let state_root = chain
+                        .get_blinded_block(&block_root)
+                        .map_err(|_| {
+                            warp_utils::reject::custom_server_error(
+                                "Error retrieving block".to_string(),
+                            )
+                        })?
+                        .map(|signed_block| signed_block.state_root())
+                        .ok_or_else(|| {
+                            warp_utils::reject::custom_not_found(
+                                "Light client bootstrap unavailable".to_string(),
+                            )
+                        })?;
+
+                    let mut state = chain
+                        .get_state(&state_root, None)
+                        .map_err(|_| {
+                            warp_utils::reject::custom_server_error(
+                                "Error retrieving state".to_string(),
+                            )
+                        })?
+                        .ok_or_else(|| {
+                            warp_utils::reject::custom_not_found(
+                                "Light client bootstrap unavailable".to_string(),
+                            )
+                        })?;
+                    let fork_name = state
+                        .fork_name(&chain.spec)
+                        .map_err(inconsistent_fork_rejection)?;
+                    let bootstrap =
+                        LightClientBootstrap::from_beacon_state(&mut state).map_err(|_| {
+                            warp_utils::reject::custom_server_error(
+                                "Failed to create light client bootstrap".to_string(),
+                            )
+                        })?;
```

Comment:
nit: returning a bootstrap upon request logic is necessary for both rest api and reqresp p2p. Should this code be moved to the chain to reduce a future diff?

Reply:
Yep, good point - we also have the same logic in p2p below, I'll move it to the `BeaconChain`.
https://github.com/sigp/lighthouse/blob/4ce01ddd11b70bf6bdd6ac3896e2b33acbf00f72/beacon_node/network/src/network_beacon_processor/rpc_methods.rs#L300

Reply:
Fixed in https://github.com/sigp/lighthouse/pull/3954/commits/7c2b5b6e997db1296e6eb98db336a5e936801b42. This actually revealed a few more issues - we are not returning `ForkVersionedResponse` (wrapper with the `version` field) in the API and also not adding context bytes to the p2p responses (will do this in a separate PR).